src/runtime/hexagon_remote/sim_remote.cpp

```diff
@@ -120,12 +175,46 @@ int halide_do_task(void *user_context, halide_task_t f, int idx,
 
 int halide_do_par_for(void *user_context, halide_task_t f,
                       int min, int size, uint8_t *closure) {
-    for (int x = min; x < min + size; x++) {
-        int result = halide_do_task(user_context, f, x, closure);
+
+    master_thread_hvx_mode = get_hvx_mode();
+    int num_worker_threads = 1;
+    if (master_thread_hvx_mode == DOUBLE_VECTOR_MODE) {
+        num_worker_threads = 3;
```

Comment:
Isn't this backwards? 2 threads for double vector mode, 3 threads for everything else?